exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -238,6 +278,11 @@ func (prwe *PRWExporter) export(ctx context.Context, tsMap map[string]*prompb.Ti
 			defer wg.Done()
 
 			for request := range input {
+				if err := ctx.Err(); err != nil {
```

Comment:
It looks a bit unexpected that we suddenly check the context Err(). Maybe add a comment to explain why this is needed.

Reply:
I switched it up a bit, but essentially I made it so that we don't have to even attempt to compose the HTTP POST request if the context has expired, by making it instead a select statement. Please take a look.